db/crud_test.go

```diff
@@ -2282,3 +2282,97 @@ func TestSyncDataCVEqual(t *testing.T) {
 		})
 	}
 }
+
+func TestProposedRev(t *testing.T) {
+
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyAll)
+
+	db, ctx := setupTestDB(t)
+	defer db.Close(ctx)
+	collection, ctx := GetSingleDatabaseCollectionWithUser(ctx, t, db)
+
+	// create 3 documents
+	body := Body{"key1": "value1", "key2": 1234}
+	_, doc1, err := collection.Put(ctx, "doc1", body)
+	require.NoError(t, err)
+	doc1Rev := doc1.GetRevTreeID()
+
+	_, doc2, err := collection.Put(ctx, "doc2", body)
+	require.NoError(t, err)
+	doc2Rev1 := doc2.GetRevTreeID()
+	_, doc2, err = collection.Put(ctx, "doc2", Body{"_rev": doc2Rev1, "key1": "value2", "key2": 5678})
+	require.NoError(t, err)
+	doc2Rev2 := doc2.GetRevTreeID()
+
+	_, doc3, err := collection.Put(ctx, "doc3", body)
+	require.NoError(t, err)
+	doc3Rev1 := doc3.GetRevTreeID()
+	_, _, err = collection.Put(ctx, "doc3", Body{"_rev": doc3Rev1, "_deleted": true})
+	require.NoError(t, err)
+
+	testCases := []struct {
+		name           string
+		revID          string
+		parentRevID    string
+		expectedStatus ProposedRevStatus
+		currentRev     string
```

Comment:
```suggestion
		expectedCurrentRev     string
```

This indicates it is the return value of the struct